lambda_function.py

```diff
@@ -1,132 +1,224 @@
-import requests
 import logging
 import os
+
 from slack_sdk import WebClient
 from slack_sdk.errors import SlackApiError
+
+from abc import ABC, abstractmethod
 from datetime import timedelta, datetime
+import requests
 
-CHANNEL_ID = os.environ.get("CHANNEL_ID")
-USER_TOKEN = os.environ.get("USER_TOKEN")
-NOTION_PAGE_ID = os.environ.get("NOTION_PAGE_ID")
-slack_client = WebClient(token=USER_TOKEN)
-
-DAILY_STATUS_FILTER_TEXT = "Good morning! Don’t forget to post your update in thread."
-DAILY_STATUS_FILTER_POD_NAME = "learningpod"
-AUTOMATED_MESSAGE = "Note: This is an automated message 😋"
-
-logger = logging.getLogger(__name__)
-
-def parse_text(title):
-    text_list = []
-    for text_data in title:
-        # Text only
-        if len(text_data) == 1:
-            text_list.append(text_data[0])
-        # Text with link
-        elif text_data[1][0] and text_data[1][0][0] == 'a':
-            # text_list.append(f"[{text_data[0]}]({text_data[1][0][1]})")
-            text_list.append(text_data[0])
-    complete_text = ''.join(text_list)
-    return complete_text
-
-def format_text(text, value_type):
-    if value_type == 'text':
-        return text
-    elif value_type == 'bulleted_list':
-        return f"•   {text}"
-    elif value_type == 'page':
-        return None
 
+class DataFetcher(ABC):
```

Comment:
can this be a more specific name? or are you planning to use this abstract class for all the case where there's a data fetching?